spec/adapters/publishing_adapter_spec.rb

```diff
@@ -537,5 +537,14 @@
 
       subject.redirect_section(section, to: "/new-location")
     end
+
+    it "redirects section via Publishing API with valid attributes" do
+      expect(publishing_api).to receive(:put_content).with(
+        redirect_content_id,
+        be_valid_against_schema("redirect")
```

Comment:
Having chatted with Chris about this, I've got a couple of ideas on how to improve matters, but I'm going to merge this as stands and then look at making an improvement to the three places where this type of matcher is being used as an argument matcher in a separate PR.